BuildForSDG/team-029-backend
# feat(auth): setup code structure and implement auth

## Description
Auth Feature Implementation
Provides the ability to register as an admin and create road wardens.
Fixes #10 
Fixes #11 
Fixes #9 

**Others**
- Setup code structure
- Remove test table from database
- Add script to create the first admin
- Setup test mocha and chi as test runner 
- Add specs

## How Has This Been Tested?
- Checkout this branch
- Run _npm install_
- Run _db-migrate up_ 
- Ensure your .env have variables similar to below: 
    export ROADRY_DEV_DATABASE_URL=postgres://moshoodsikiru:root@localhost:5432/roadry
    export JWT_SECRET_KEY=flajfal41h5j4k2l4j5345@
- Bind your environment variable to the process by running _source .env_ in your terminal
- Create the first admin to be able to use the registration route
- Run _npm run create-admin_
- Run _npm start_
- Make a POST request to { base_uri }/api/v1/auth/login with the admin details you created
- Expect a successful login 
- To start creating new admins and road wardens
- Make a POST request to { base_uri} /api/v1/auth/register with **authorization**  header set to the JWT token you got on logging in using sample body: {
       "first_name": "Lucky",
        "last_name": "Temple",
        "phone_number": "12345678910",
        "office_address": "New York",
        "user_type": "admin",
        "password":"password",
        "confirm_password": "password",
         "email": "templemen@enyata.com"
}
- Expect a successful response and new user should now be able to login


## Checklist:

- [ x ] My code follows the style guidelines of this project
- [ x ] I have performed a self-review of my own code
- [ x ] I have commented my code, particularly in hard-to-understand areas
- [ x ] I have added necessary inline code documentation
- [ x ] I have added tests that prove my fix is effective and that this feature works
- [ x ] New and existing unit tests pass locally with my changes
